src/Interpreters/ITokenExtractor.h

```diff
@@ -160,48 +160,58 @@ struct NgramTokenExtractor final : public ITokenExtractorHelper<NgramTokenExtrac
     bool nextInString(const char * data, size_t length, size_t * __restrict pos, size_t * __restrict token_start, size_t * __restrict token_length) const override;
-
     bool nextInStringPadded(const char * data, size_t length, size_t * __restrict pos, size_t * __restrict token_start, size_t * __restrict token_length) const override;
-
     bool nextInStringLike(const char * data, size_t length, size_t * __restrict pos, String & token) const override;
-
     void substringToBloomFilter(const char * data, size_t length, BloomFilter & bloom_filter, bool is_prefix, bool is_suffix) const override;
-
     void substringToGinFilter(const char * data, size_t length, GinFilter & gin_filter, bool is_prefix, bool is_suffix) const override;
 };
 
-/// Parser doing "no operation". Returns the entire input as a single token.
-struct NoOpTokenExtractor final : public ITokenExtractorHelper<NoOpTokenExtractor>
+/// Parser extracting tokens which are separated by specific chacters.
+/// Result is slightly different than for SplitTokenExtractor.
+struct LogTokenExtractor final : public ITokenExtractorHelper<SplitTokenExtractor>
 {
-    static const char * getName() { return "noop"; }
+    static const char * getName() { return "log"; }
     static const char * getExternalName() { return getName(); }
 
-    std::vector<String> getTokens(const char * data, size_t length) const override;
+    /// List of separators taken from BigQuery's LOG_ANALYZER.
+    /// Ordered by descending length, so matching prefers longer separators (e.g. in '%21xyz' both '%' and '%21' match)
+    static inline std::vector<String> separators =
+        {"%2520", "%2526", "%21", "%26", "%3B", "%3b", "%7C", "%7c", "%20", "%2B", "%2b", "%3D", "%3d",
+         "%5D", "%5d", "%5B", "%5b", "%3A", "%3a", "%0A", "%0a", "%2C", "%2c", "%28", "%29",
+         "{", "}", "<", ">", "(", ")", "{", "}", "|", "!", ";", ",", "\'", "\"", "*", "&", "?", "+",
+         "/", ":", "=", "@", ".", "-", "$", "%", "\\", "_", "\n", "\r", "\t", };
 
+    std::vector<String> getTokens(const char * data, size_t length) const override;
     bool nextInString(const char * data, size_t length, size_t * __restrict pos, size_t * __restrict token_start, size_t * __restrict token_length) const override;
-
     bool nextInStringLike(const char * data, size_t length, size_t * __restrict pos, String & token) const override;
 };
 
+/// Parser doing "no operation". Returns the entire input as a single token.
+struct NoOpTokenExtractor final : public ITokenExtractorHelper<NoOpTokenExtractor>
+{
+    static const char * getName() { return "no_op"; }
+    static const char * getExternalName() { return getName(); }
+
+    std::vector<String> getTokens(const char * data, size_t length) const override;
+    bool nextInString(const char * data, size_t length, size_t * pos, size_t * token_start, size_t * token_length) const override;
+    bool nextInStringLike(const char * data, size_t length, size_t * pos, String & token) const override;
```

Comment:
The function is inherited from `ITokenExtractor`, where we have the `__restrict` keyword. So I would either remove it from everywhere or keep it everywhere to make sure the function are aligned.

Reply:
You are right, fixed.